crates/core/src/host/wasmer/mod.rs

```diff
@@ -1,22 +1,40 @@
+use std::borrow::Cow;
 use std::sync::Arc;
 
-use wasmer::wasmparser::Operator;
-use wasmer::{AsStoreRef, CompilerConfig, EngineBuilder, Memory, MemoryAccessError, Module, RuntimeError, WasmPtr};
-use wasmer_middlewares::Metering;
+use anyhow::Context;
+use once_cell::sync::Lazy;
+use wasmtime::{Engine, Linker, Module, StoreContext, StoreContextMut};
 
 use crate::database_instance_context::DatabaseInstanceContext;
 use crate::error::NodesError;
 use crate::hash::Hash;
 
-mod opcode_cost;
 mod wasm_instance_env;
 mod wasmer_module;
 
 use wasmer_module::WasmerModule;
 
+use self::wasm_instance_env::WasmInstanceEnv;
+
 use super::scheduler::Scheduler;
+use super::wasm_common::module_host_actor::InitializationError;
 use super::wasm_common::{abi, module_host_actor::WasmModuleHostActor, ModuleCreationError};
-use super::{EnergyMonitor, EnergyQuanta};
+use super::EnergyMonitor;
+
+static ENGINE: Lazy<Engine> = Lazy::new(|| {
+    let mut config = wasmtime::Config::new();
+    config
+        .cranelift_opt_level(wasmtime::OptLevel::Speed)
+        .consume_fuel(true)
```

Comment:
In the future, I think we should investigate whether epoch interruption is sufficient for our needs. Within this PR, fuel seems like the right thing to use, as it's the closest translation of Wasmer metering.

Reply:
Would you mind filing a ticket about this?